runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java

```diff
@@ -67,6 +69,9 @@ public class Instance {
     private final ExecutionListener listener;
     private final Exports fluentExports;
 
+    private final Optional<BiFunction<Instance, Integer, List<StackTraceElement>>>
```

Comment:
I extracted all the callback logic into this callback, we can move to a class or interface if preferable.